tests/PromiseExceptionTest.php

```diff
@@ -12,9 +12,9 @@ class PromiseExceptionTest extends \PHPUnit_Framework_TestCase
 {
     public function testGetException()
     {
-        $request = $this->getMock('Psr\Http\Message\RequestInterface');
-        $response = $this->getMock('Psr\Http\Message\ResponseInterface');
-        $promise = $this->getMock('GuzzleHttp\Promise\PromiseInterface');
+        $request = $this->getMockBuilder('Psr\Http\Message\RequestInterface')->getMock();
+        $response = $this->getMockBuilder('Psr\Http\Message\ResponseInterface')->getMock();
+        $promise = $this->getMockBuilder('GuzzleHttp\Promise\PromiseInterface')->getMock();
```

Comment:
do we need backwards compatitility with old phpunit versions? the new way would be createMock when you do not need to customize the mock builder... i'd find it simpler to add phpunit to the require-dev and write tests against a specific phpunit version. with phpunit 6, there will be namespaces and all tests would again fail if we don't adjust.